Readme.md

```diff
@@ -222,60 +222,113 @@ if(maybeABasket.isNothing()) {
 ## Maybe api
 
 - Maybe take a value and return a maybe, if the passed value is null or undefined, all operation on the maybe to change the value will be ignored.
+
 - Maybe.of: same as Maybe
-- Maybe.lift: Take a function and wrap its return value in Maybe
+
+- Maybe.lift: Takes a function and wrap its return value in Maybe
 ```ts
-lift<A, B>(fn: (v: A) => B): (v: A): Maybe<B>;
+// Maybe.lift(fn: A => B): (v: A) => Maybe<B>;
+const fn = Maybe.lift(v => v * 2);
+// is the same as
+const fn = v => Maybe.of(v * 2);
 ```
+
 - maybe.map: Take a function and apply it to the value if there is one
 ```ts
-map<A, B>(this: Maybe<A>, fn: (value: A) => B): Maybe<B>;
-map<A, B>(this: Maybe<nothing>, fn: (value: A) => B): Maybe<nothing>;
+// Maybe<A>.map<fn: A => B>: Maybe<B>;
+Maybe.of(5).map(v => v * 2);
+-> Maybe<10>;
+// Maybe<null | undefined>.map(fn: A => B): Maybe<null | undefined>;
+Maybe.of(null).map(v => v * 2);
+-> Maybe<null>;
 ```
 - maybe.flatten: If the value is another Maybe merge the two together. If there is no value, do nothing.
 ```ts
-flatten<A>(this: Maybe<Maybe<A>>): Maybe<A>;
-flatten<A>(this: Maybe<nothing>): Maybe<nothing>;
+// Maybe<Maybe<A>>.flatten(): Maybe<A>;
+Maybe.of(Maybe.of(5)).flatten();
+-> Maybe<5>;
+// Maybe<null | undefined>.flatten(): Maybe<null | undefined>;
+Maybe.of(null).flatten();
+-> Maybe<null>;
 ```
 - maybe.chain
 ```ts
-chain<A, B>(this: Maybe<A>, fn: ((v: A) => Maybe<B>)): Maybe<B>;
-chain<A, B>(this: Maybe<nothing>, fn: ((v: A) => Maybe<B>)): Maybe<nothing>;
+//Maybe<A>.chain(fn: A => Maybe<B>): Maybe<B>;
+Maybe.of(5).chain(v => Maybe.of(v * 2));
+-> Maybe<10>;
+// Maybe<nothing>.chain(fn: A => Maybe<B>): Maybe<nothing>;
+Maybe.of(null).chain(v => Maybe.of(v * 2));
+-> Maybe<null>;
 ```
 - maybe.ap
 ```ts
-ap<A, B>(this: Maybe<(v: A) => B>, v: Maybe<A>): Maybe<B>;
-ap<A, B>(this: Maybe<nothing>, v: Maybe<A>): Maybe<nothing>;
-ap<A, B>(this: Maybe<(v: A) => B>, v: Maybe<nothing>): Maybe<nothing>;
+// Maybe<A => B>.ap(other: Maybe<A>): Maybe<B>;
+Maybe.of(v => v * 2).ap(Maybe.of(5));
+-> Maybe<10>;
+// Maybe<nothing>.ap(other: Maybe<A>): Maybe<nothing>;
+Maybe.of(null).ap(Maybe.of(5));
+-> Maybe<null>;
+// Maybe<A => B>.ap(other: Maybe<nothing>): Maybe<nothing>;
+Maybe.of(v => v * 2).ap(Maybe.of(null));
+-> Maybe<null>;
 ```
 - maybe.sequence: when the maybe hold an applicative, it will swap the maybe with the applicative. If the maybe hold nothing it will place the maybe inside an apllicative.
```

Comment:
- maybe.sequence: when the maybe holds an applicative, it will swap the maybe with the applicative. When the maybe does not hold anything, it will place the maybe inside an applicative.